### PR TITLE
5345 manual reqs stock on hand value overwritten

### LINKED ISSUE #5345: Manual reqs: Stock on hand value overwritten
## What went wrong? 😲

The stock on hand value is overwritten when you enter a value for another input.

Side quest: can you focus the stock on hand input when the page loads please

Here's the issue:

https://github.com/user-attachments/assets/fc33ad80-3433-4a1c-966d-d8c037d339c1

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Go to an item line in a manual requisition
2. enter a value into stock on hand
3. press TAB
4. enter a value into Stock arriving
5. press TAB
6. the stock on hand is changed to 0

## Your environment 🌱

- Open mSupply Version:
- Legacy mSupply Central Server Version:

- Platform:

- Database type:

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #5345

# 👩🏻‍💻 What does this PR do?
Separate var for stock on hand and initial stock on hand, also autofocus on first input on load.

# 🧪 Testing

- [ ] Add some lines to a general manual requisition
- [ ] Go to line
- [ ] Enter value in stock on hand -> TAB enter value -> TAB enter value
- [ ] Stock on hand shouldn't change to 0

# 📃 Documentation

- [x] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

**Comment by mark-prins** [comment]
the overwriting is fixed! 🙏 

I have spotted another minor thing - perhaps could be fixed in the same PR?

Two things in the stats popup:
* the bar no longer needs the numbers inside (the white text)
* the width of the bars is the same for each type, but they should be relative to the number they represent

Screenshot 2024-11-07 at 3 34 40 PM

( Checking the code in `ResponseStoreStats` I can see that the top bar is using `predictedStockPercent` for all three bars, and the lower uses `requestedPercent` for both bars )

**Review by mark-prins — APPROVED** [review]
👍  LGTM! thanks - have spotted another minor UI thing, happy to re-look if you want to add on to this PR rather than make a new issue/pr

**Comment by roxy-dao** [comment]
I'll add to this PR and ask for a re-review when done! Thanks Mark :D

**Comment by roxy-dao** [comment]
> 7b6dbdf

File: client/packages/common/src/ui/components/charts/NewValueBar/NewValueBar.tsx [hunk 1]
```diff
@@ -1,43 +1,27 @@
...
-            ) : null}
-          </Box>
-        </Box>
+      <Box flexBasis={`${flexBasis}%`} flexGrow={1}>
+        <Box sx={{ backgroundColor: colour, height: '20px' }}></Box>
+        <Box
+          style={{
+            paddingRight: 10,
+            paddingTop: 10,
+          }}
```
Comment (roxy-dao) [inline client/packages/common/src/ui/components/charts/NewValueBar/NewValueBar.tsx:23]: If the value is too small it won't show without this 😓

**Comment by mark-prins** [comment]
Looks great! thanks @roxy-dao 🙏

**Comment by alainsussol** [comment]
# 🧪 Testing
Tested with v2.4.0 on tablet Android 4


- [ ] Add some lines to a general manual requisition
- [ ] Go to line
- [ ] Enter value in stock on hand -> TAB enter value -> TAB enter value
- [ ] Stock on hand shouldn't change to 0 **-PASS-**

**Pull request merged**